spec/OneClickInstall_spec.rb

```diff
@@ -52,13 +62,100 @@
   describe "#SetRequiredRepository" do
     context "set repository to recommended" do
       it "Ensures that the repository with the specified URL is selected for addition" do
-        url = "http://download.opensuse.org/tumbleweed/repo/oss/"
-        subject.instance_variable_set(:@repositories, { url => { "name" => "vim" , "recommended" => "false" }})
-        subject.SetRequiredRepository(url)
+        subject.instance_variable_set(:@repositories, { urls[0] => { "name" => "oss" , "recommended" => "false" }})
```

Comment:
If I understand what you mean here, that would save us from setting the also name before every example, right? I think the benefit doesn't worth to increase the complexity and decrease the readability, WDYT?

Reply:
yes, I didn't test the code, but if you can pass and object, we could just pass the right object, something like:
subject.instance_variable_set(:@repositories.urls[0].recommended, "false" }}).
That is what I thought at the beginning that it would have been better choice to use a hash instead of array for those urls, so it would be more readable, the only thing is the in the method where you really pass the array (somewhere below), in that case you just need to convert the hash to array, but for the rest of the code you can avoid index. In that case this way would be more clear, but it is just a suggestion.

Reply:
Ah wait, now I got what you mean. It doesn't really matter what !let does here because "instance_variable_set" overwrites @repositories.

Reply:
yes, so we might not need to set the whole thing twice, only what changes and for being more readable have a hash instead of an array.

Reply:
I am not sure if I get what you mean. 
subject.instance_variable_set(:@repositories.urls[0].recommended, "false" }}) 
would erase repositories[urls[0][name] for example, so I don't think is appropriate here, unless I remove the check for the name.

Reply:
yes, instance_variable_set will overwrite the whole content of the variable